File: January 2023/Day 25/Type it!/type-it.cpp
//{ Driver Code Starts
// Initial template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function template for C++

class Solution {
  public:
    // Time Complexity :- O(n^2);
    // Space Complexity :- O(1);
    int minOperation(string s) {
        // code here
        int ans=s.length();
        for(int i=0;i<s.length()/2;i++) {
            bool isValid=true;
            for(int j=0;j<=i && isValid;j++) {
                if(s[j]!=s[j+i+1]) {
                    isValid=false;
                }
            }
            if(isValid) ans = s.length()-i;
        }
        return ans;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        string s;
        cin >> s;
        Solution ob;
        cout << ob.minOperation(s) << "\n";
    }
    return 0;
}

// } Driver Code Ends

File: January 2023/Day 25/Type it!/type-it.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;
class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out=new PrintWriter(System.out);
        int t = Integer.parseInt(read.readLine());

        while (t-- > 0) {
            String s = read.readLine().trim();
            Solution ob = new Solution();
            int ans = ob.minOperation(s);
            out.println(ans);
        }
        out.close();
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    // Time Complexity :- O(n);
    // Space Complexity :- O(n);
    int minOperation(String s) {
        // code here
        HashSet<String> set = new HashSet<>();
        for(int i=0;i<s.length()/2;i++) set.add(s.substring(i+1,2*(i+1)));
        
        int ans=s.length();
        for(int i=0;i<s.length()/2;i++) {
            if(set.contains(s.substring(0,i+1))) ans=s.length()-i;
        }
        
        return ans;
    }
}
